src/Swashbuckle.AspNetCore.Cli/Program.cs

```diff
@@ -165,14 +172,22 @@ public static int Main(string[] args)
                         }
                     }
 
-
                     using Stream stream = outputViaConsole ? Console.OpenStandardOutput() : File.Create(outputPath);
-                    using StreamWriter writer = new StreamWriter(stream, outputViaConsole ? Console.OutputEncoding : Encoding.UTF8);
+                    using StreamWriter writer = new(stream, outputViaConsole ? Console.OutputEncoding : Encoding.UTF8);
 
-                    writer.WriteLine($"Swagger Document Names (Surrounded by single quotes):");
-                    foreach (var name in docNames)
+                    if (docMetaProvider == null)
                     {
-                        writer.WriteLine($"'{name}'");
+                        writer.WriteLine($"ERROR: ISwaggerProvider instance DOES NOT support ISwaggerDocumentMetadaatProvider, so this coommand "
+                            + "DOES NOT support listing out the document names. Try updating the version of Swashbuckle used by the project you are examining to a newer version.");
```

Comment:
A few comments on this message:

1. Don't use capitalisation to emphasise words - it just reads like it's shouting at the user.
2. There's a number of typos.
3. Use `nameof()` to format in type names, then they'll definitely be spelled correctly.
4. I'm not sure that such a scenario would manifest that way. The tool is loading the app's types to inspect them with reflection, so if it's our provider it must implement the interface by definition. If it's a custom user-provider implementation, then upgrading won't help. I would instead just state the fact that the resolved provider doesn't implement the interface (maybe print the runtime type's name to aid with debugging) and leave it at that.

Reply:
I thnk I've fixed this as well.